horreum-backend/src/main/java/io/hyperfoil/tools/horreum/mapper/ExperimentProfileMapper.java

```diff
@@ -48,7 +54,10 @@ public static ExperimentProfileDAO to(ExperimentProfile dto) {
         ep.selectorFilter = dto.selectorFilter;
 
         ep.comparisons = dto.comparisons.stream().map(ExperimentProfileMapper::toExperimentComparison).collect(Collectors.toList());
-
+        if (dto.testId != null) {
+            em = Arc.container().instance(EntityManagerProvider.class).get().getEntityManager();
+            ep.test = em.getReference(TestDAO.class, dto.testId);
+        }
```

Comment:
The mappers are static property mappers, relationships between entities should be performed within the managed beans.

The custom `EntityManagerProvider` should be removed and instead a single line in `ExperimentServiceImpl.importTest()` on line 313 should be added;

```
profile.test = em.getReference(TestDAO.class, ep.testId);
```

Reply:
sure